src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1396,35 +1399,65 @@ private async Task WriteStreamPropertiesAsync(SelectExpandNode selectExpandNode,
         {
             ISet<IEdmNavigationProperty> navigationProperties = selectExpandNode.SelectedNavigationProperties;
 
-            if (navigationProperties != null)
+            if (navigationProperties == null)
             {
-                IEnumerable<string> changedProperties = null;
-                if (null != resourceContext.EdmObject && resourceContext.EdmObject is IDelta changedObject)
+                yield break;
+            }
+
+            IEnumerable<string> changedProperties = null;
+
+            if (resourceContext.EdmObject != null && resourceContext.EdmObject is IDelta changedObject)
```

Comment:
```suggestion
            if (resourceContext.EdmObject is IDelta changedObject)
```
Since `resourceContext.EdmObject is IDelta` would evaluate to false if `resourceContext.EdmObject` is `null` - I BELIEVE!